crates/sargon/src/signing/host_interaction/sign_response.rs

```diff
@@ -6,18 +6,18 @@ use crate::prelude::*;
 /// Radix network.
 #[derive(Clone, PartialEq, Eq, derive_more::Debug)]
 #[debug("SignResponse {{ signatures: {:#?} }}", signatures.values().map(|f| format!("{:#?}", f)).join(", "))]
-pub struct SignResponse {
-    pub signatures: IndexMap<FactorSourceIDFromHash, IndexSet<HDSignature>>,
+pub struct SignResponse<S: Signable> {
```

Comment:
Looks like this can be made generic over `T: SignableID` only, not the whole `Signable` trait?